CMakeLists.txt

```diff
@@ -246,6 +247,8 @@ add_definitions (-D_GNU_SOURCE)
 add_definitions (-D_BSD_SOURCE)
 add_definitions (-D_DEFAULT_SOURCE)
 
+add_subdirectory (src)
+
```

Comment:
It seems my suggestion for moving this was not quite right, and is the cause of the `make-release-archive` task failing.

`src/CMakeLists.txt` makes the assumption that `SOURCE_DIR` is defined. But it is defined below on line 260. So it ends up being empty (because cmake, like bash, treats unset variables like empty strings) causing that error when producing the dist tarball. Also, `src/CMakeLists.txt` has comments indicating it assumes other directories have been included.

My new theory:
- Move this `add_subdirectory (src)` back.
- Add `add_subdirectory (src/common)` here, so the common-config.h.in is always configured. This handles the previous include errors when building only libbson with `-DENABLE_MONGOC=OFF`
- Remove `add_subdirectory (common)` from src/CMakeLists.txt

The problematic tasks were tasks that were building with `-DENABLE_MONGOC=OFF` and tasks that were building the release tarball. You can test both of these locally:

To build only libbson:
```
mkdir cmake-build-bsononly
cd cmake-build-bsononly
cmake -DENABLE_MONGOC=OFF ..
cmake --build . --target bson_shared
```

To build the distribution tarball:
```
mkdir cmake-build-dist
cd cmake-build-dist
cmake -DENABLE_BSON=ON -DENABLE_MAN_PAGES=ON -DENABLE_HTML_DOCS=ON ..
cmake --build . --target dist
```

Aside: cmake docs say [SOURCE_DIR is a built-in read-only variable](https://cmake.org/cmake/help/latest/prop_tgt/SOURCE_DIR.html). So that seems fishy... but a problem for another time. Updating our cmake scripts can be quite frustrating sometimes.